charts/redpanda/testdata/template-cases.txtar

```diff
@@ -948,3 +948,174 @@ statefulset:
 # ASSERT-FIELD-EQUALS ["apps/v1/StatefulSet", "default/also-not-redpanda", "{.spec.template.spec.containers[*].name}", ["redpanda", "sidecar"]]
 nameOverride: "not-redpanda"
 fullnameOverride: "also-not-redpanda"
+
+-- trust-store-reference --
+# ASSERT-NO-ERROR
+# ASSERT-GOLDEN
```

Comment:
Could we add in a `ASSERT-FIELD-EQUALS` or `ASSERT-FIELD-CONTAINS` here? This test won't effectively prevent regressions without some assertion to showcase what shouldn't regress.

You might want to add in a new assertion type, maybe `ASSERT-NESTED-FIELD-EQUALS`, so we can effectively assert against the contents of `redpanda.yaml`

Reply:
Done, but I would love you to review my proposal of nested field equal as I think it will be only applicable to string that the content have json object.

Reply:
LGTM! assertions against redpanda.yaml is likely ~75% of our use case for this assertion. If need be we can update it later on to handle nest YAML and JSON but this is great for now.